test/fft_test.rg

```diff
@@ -1,108 +1,409 @@
--- Copyright 2020 Stanford University
---
--- Licensed under the Apache License, Version 2.0 (the "License");
--- you may not use this file except in compliance with the License.
--- You may obtain a copy of the License at
---
---     http://www.apache.org/licenses/LICENSE-2.0
---
--- Unless required by applicable law or agreed to in writing, software
--- distributed under the License is distributed on an "AS IS" BASIS,
--- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
--- See the License for the specific language governing permissions and
--- limitations under the License.
-
 import "regent"
 
 local fft = require("fft")
 
 local cmapper = require("test_mapper")
+local format = require("std/format")
+
+--Import cuFFT API
+local cufft_c = terralib.includec("cufftXt.h")
+terralib.linklibrary("libcufft.so")
+
+----PRINT FUNCTIONS-----
+
+-- Task to print out input or output array with fieldspace complex64. Takes a region and a string representing the name of the array
+__demand(__inline, __leaf)
+task print_array_double_complex(input : region(ispace(int1d), complex64), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  for x in input do
+    var currComplex = input[x]
+    format.println("{} + {}j, ", currComplex.real, currComplex.imag)
+  end
+  format.println("]\n")
+end
+
+-- Task to print out input or output array with fieldspace complex32. Takes a region and a string representing the name of the array
+__demand(__inline, __leaf)
+task print_array_float_complex(input : region(ispace(int1d), complex32), arrayName: rawstring)
```

Comment:
Overall this seems like a lot of redundant code that we could unify through some metaprogramming.

Reply:
And if we need to support complex types in `format`, let's do that.

Reply:
On my to-do list